### PR TITLE
1860 check cluster status

### LINKED ISSUE #1860: API checks if a cluster has been initialised.
**User Story**
As a user of kubermatic API server I'd like it to detect and signal back when I'm trying to do operations associated with `cluster` resource that hasn't been initialised yet. For example adding a `node` to not initialised `cluster` will return an error. The actual error can be misleading for example it can return `404` `secret \"admin-kubeconfig\" not found`. Instead our API should check if the `cluster` resource has been initialised and if not it should yield back an error. Possibles error codes could be `202` or `503` or something else ? 

How to check if a cluster has been initialised ?
use `cluster/health` endpoint and check if all listed components have its field set to `true`

Possible implementation:
Change `GET` method of the rbac compliant cluster provider so that it accepts "options" (similarly to List method). If the option is set then the provider will ensure that the cluster was initialised and if not it will return an error. 
Next identify all possible places in API. For example  the handler for `POST` to `/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodes` and set appropriate option when getting the cluster.

**Acceptance criteria**
- API verifies if the given cluster has been initialised before allowing to an operation that requires it.

### PR DESCRIPTION
**Pull request by glower**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1860

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/provider/kubernetes/rbac-compliant-cluster.go [hunk 1]
```diff
@@ -138,12 +138,27 @@ func (p *RBACCompliantClusterProvider) List(project *kubermaticapiv1.Project, op
...
+		return nil, err
+	}
+	if options.CheckInitStatus {
+		isHealthy := cluster.Status.Health.Apiserver &&
+			cluster.Status.Health.Scheduler &&
+			cluster.Status.Health.Controller &&
+			cluster.Status.Health.MachineController &&
+			cluster.Status.Health.Etcd
+		if !isHealthy {
+			return nil, kerrors.NewServiceUnavailable("Cluster components are not ready now")
```
Comment (mrIncompetent) [inline api/pkg/provider/kubernetes/rbac-compliant-cluster.go]: `Cluster components are not ready yet`

**Comment by mrIncompetent** [comment]
Code looks good, but the dep change seems odd. 
@glower Are you using the correct dep version?

**Comment by glower** [comment]
* fixed godep
* fixed error message
@mrIncompetent if all good, can you please merge it for me. I will check what is wrong with my dep

**Comment by mrIncompetent** [comment]
@glower tests are failing

**Comment by glower** [comment]
all tests fixed

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**